scrapy/extensions/feedexport.py

```diff
@@ -187,28 +187,45 @@ def _store_in_thread(self, file):
 
 
 class SpiderSlot(object):
-    def __init__(self, file, exporter, storage, uri):
+    def __init__(self, file, exporter, storage, uri, feed_format):
         self.file = file
         self.exporter = exporter
         self.storage = storage
         self.uri = uri
+        self.feed_format = feed_format
         self.itemcount = 0
+        self._exporting = False
 
 
 class FeedExporter(object):
 
     def __init__(self, settings):
         self.settings = settings
-        self.urifmt = settings['FEED_URI']
-        if not self.urifmt:
+
+        if not settings['FEED_URI']:
             raise NotConfigured
-        self.format = settings['FEED_FORMAT'].lower()
+
+        # Begin: Backward compatibility for string values in the FEED_URI setting
+        if isinstance(settings['FEED_URI'], dict):
+            self.urifmts = {uri.strip(): feed_format.strip().lower()
```

Comment:
I think we should do something if `FEED_FORMAT` is used in combination with a `dict` value for `FEED_URI`.

Maybe we could log the warning below as long as `FEED_FORMAT` format is used, regardless or the value of `FEED_URI`?

Reply:
I see your point, but `FEED_FORMAT` has a default value of `jsonlines` in `scrapy/settings/default_settings.py`. I'd be ok with setting that to `None`, but it would technically be a backward imcompatible change. What do you think?

Reply:
I think we should remove `FEED_FORMAT` from that file (in line with how [`SPIDER_MANAGER_CLASS` was renamed there in the past](https://github.com/scrapy/scrapy/commit/403e7c7c7059486c5a4c59f900a65378de2ba3d2#diff-ff579a5763caae1855c840beac29a49b), but this time simply removing), and as long as you replace `settings['FEED_FORMAT'].lower()` with `settings.get('FEED_FORMAT', 'jsonlines').lower()` below, it should be backward compatible.

Reply:
Updatedm althought in a slightly different wasy as you suggested. Please check again, thanks.